server/checks/Approval.js

```diff
@@ -13,17 +14,123 @@ export default class Approval extends Check {
+                                .reduce((result, approvalGroup) => {
+                                  const needed = groups[approvalGroup].minimum
+                                  const given = approvals.groups[approvalGroup]
+                                  const diff = needed - given
+                                  if (diff > 0) {
+                                    result.push({approvalGroup, diff, needed, given})
+                                  }
+                                  return result
+                                }, [])
+
+      if (unsatisfied.length > 0) {
+        const firstUnsatisfied = unsatisfied[0]
+        return {
+          description: `This PR misses ${firstUnsatisfied.diff} approvals from group ${firstUnsatisfied.approvalGroup} (${firstUnsatisfied.given}/${firstUnsatisfied.needed}).`,
+          state: 'pending',
+          context
+        }
+      }
+    }
+
+    if (approvals.total < minimum) {
+      return {
+        description: `This PR needs ${minimum - approvals.total} more approvals (${approvals.total}/${minimum} given).`,
+        state: 'pending',
+        context
+      }
+    }
+
+    return {
+      description: `This PR has ${approvals.total}/${minimum} approvals since the last commit.`,
+      state: 'success',
+      context
     }
-    return `This PR has ${actual}/${needed} approvals since the last commit.`
   }
 
-  static async countApprovals(github, repository, comments, config, token) {
+  /**
+   * Returns the comment if it matches the config, null otherwise.
+   */
+  static async doesCommentMatchConfig(github, repository, comment, fromConfig, token) {
+    // persons must either be listed explicitly in users OR
+    // be a collaborator OR
+    // member of at least one of listed orgs
+    const {orgs, collaborators, users} = fromConfig
+    const username = comment.user.login
+    const {full_name} = repository
+    // first do the quick username check
+    if (users && users.indexOf(username) >= 0) {
+      debug(`${full_name}: ${username} is listed explicitly`)
+      return comment
+    }
+    // now collaborators
+    if (collaborators) {
+      const isCollaborator = await github.isCollaborator(repository.owner.login, repository.name, username, token)
+      if (isCollaborator) {
+        debug(`${full_name}: ${username} is collaborator`)
+        return comment
+      }
+    }
+    // and orgs
+    if (orgs) {
+      const orgMember = await Promise.all(orgs.map(o => github.isMemberOfOrg(o, username, token)))
+      if (orgMember.indexOf(true) >= 0) {
+        debug(`${full_name}: ${username} is org member`)
+        return comment
+      }
+    }
+    debug(`${full_name}: ${username}'s approval does not count`)
+    // okay, no member of anything
+    return null
+  }
 
-    const {pattern, ignore} = config
+  static async getApprovalsForConfig(github, repository, comments, config, token) {
+    const that = this
+
+    async function checkComment(stats, comment) {
```

Comment:
We could use `const checkComment = async (stats, comment) => { ... }` to avoid the `that = this`.